File: xsolla-store-sdk/src/main/java/com/xsolla/android/store/entity/response/order/OrderResponse.java
package com.xsolla.android.store.entity.response.order;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class OrderResponse {

    @SerializedName("order_id")
    private int orderId;
    private Status status;
    private Content content;

    public int getOrderId() {
        return orderId;
    }

    public Status getStatus() {
        return status;
    }

    public Content getContent() {
        return content;
    }

    public enum Status {
        @SerializedName("new") NEW,
        @SerializedName("paid") PAID,
        @SerializedName("done") DONE,
        @SerializedName("canceled") CANCELED
    }

    private class Content {
        private Price price;
        @SerializedName("virtual_price")
        private VirtualPrice virtualPrice;
        @SerializedName("is_free")
        private boolean isFree;
        private List<Item> items;
    }

    private class Item {
        private String sku;
        private long quantity;
        @SerializedName("is_free")
        private boolean isFree;
        private Price price;
    }

    private class Price {
        private String amount;
        @SerializedName("amount_without_discount")
        private String amountWithoutDiscount;
        private String currency;
    }

    private class VirtualPrice {
        private long amount;
        @SerializedName("amount_without_discount")
        private String amountWithoutDiscount;
        private String currency;
    }

}

File: xsolla-store-sdk/src/main/java/com/xsolla/android/store/entity/response/order/OrderResponse.kt
package com.xsolla.android.store.entity.response.order

import com.google.gson.annotations.SerializedName

data class OrderResponse(
    @SerializedName("order_id")
    val orderId: Int,
    val status: Status? = null,
    val content: Content? = null
) {
    enum class Status {
        @SerializedName("new")
        NEW,
        @SerializedName("paid")
        PAID,
        @SerializedName("done")
        DONE,
        @SerializedName("canceled")
        CANCELED
    }

    data class Content(
        val price: Price? = null,
        @SerializedName("virtual_price")
        val virtualPrice: VirtualPrice? = null,
        @SerializedName("is_free")
        val isFree: Boolean,
        val items: List<Item> = emptyList()
    )

    data class Item(
        val sku: String? = null,
        val quantity: Long,
        @SerializedName("is_free")
        val isFree: Boolean,
        val price: Price? = null
    )

    data class Price(
        val amount: String? = null,
        @SerializedName("amount_without_discount")
        val amountWithoutDiscount: String? = null,
        val currency: String? = null
    )

    data class VirtualPrice(
        val amount: Long,
        @SerializedName("amount_without_discount")
        val amountWithoutDiscount: String? = null,
        val currency: String? = null
    )
}

data class WsOrderResponse(
    @SerializedName("order_id")
    val orderId: Int,
    val status: OrderResponse.Status
)